sentinel/src/lib/validation.js

```
@@ -1,290 +1,356 @@
+      const fields = _.clone(validation.funcArgs);
+      const duration = _parseDuration(fields.pop());
+      const startDate = moment()
+        .subtract(duration)
+        .valueOf();
+      _exists(doc, fields, { startDate: startDate }, (err, result) => {
+        if (err) {
+          logger.error('Error running "uniqueWithin" validation', err);
+        }
+        callback(err, !result);
+      });
+    },
+    exists: (doc, validation, callback) => {
+      const formName = validation.funcArgs[0];
+      const fieldName = validation.funcArgs[1];
+      _exists(
+        doc,
+        [fieldName],
+        { additionalFilter: `form:${formName}` },
+        (err, result) => {
+          if (err) {
+            logger.error('Error running "exists" validation', err);
+          }
+          callback(err, result);
+        }
+      );
+    },
+    /*
+         * Check if the week is a valid ISO week given a year.
+         */
+    isISOWeek: (doc, validation, callback) => {
+      const weekFieldName = validation.funcArgs[0];
+      const yearFieldName =
+        validation.funcArgs.length > 1 ? validation.funcArgs[1] : null;
+      if (
+        !_.has(doc, weekFieldName) ||
+        (yearFieldName && !_.has(doc, yearFieldName))
+      ) {
+        callback(
+          new Error('Validation failed: input field(s) do not exist'),
+          false
+        );
+      } else {
+        const year = yearFieldName
+          ? doc[yearFieldName]
+          : new Date().getFullYear();
+        const isValidISOWeek =
+          /^\d{1,2}$/.test(doc[weekFieldName]) &&
+          /^20\d{2}$/.test(year) &&
+          doc[weekFieldName] >= 1 &&
+          doc[weekFieldName] <= moment(`${year}-01-01`).isoWeeksInYear();
+        if (isValidISOWeek) {
+          callback(null, true);
+        } else {
+          callback(
+            new Error(
+              'Validation failed: the number of week is greater than the maximum'
```

As above, this error will be lost in oblivion. Consider logging this instead.